lib/src/client.dart

```diff
@@ -1702,8 +1707,15 @@ class Client extends MatrixApi {
       });
       _currentSyncId = syncRequest.hashCode;
       onSyncStatus.add(SyncStatusUpdate(SyncStatus.waitingForResponse));
-      final syncResp =
-          await syncRequest.timeout(timeout + const Duration(seconds: 10));
+
+      // The timeout for the response from the server. If we do not set a sync
+      // timeout (for initial sync) we give the server a longer time to
+      // responde.
+      final responseTimeout = timeout.inMilliseconds == 0
+          ? const Duration(minutes: 2)
+          : timeout + const Duration(seconds: 10);
+
+      final syncResp = await syncRequest.timeout(responseTimeout);
```

Comment:
Do we even need a timeout for initial sync? It will fail eventually anyway, if there was a network error or such, right?

Reply:
No, it is possible that it loads forever with the http package. Happens from time to time